documents/Secure_Tunnel_Userguide.md

```diff
@@ -227,3 +227,8 @@ message->withPayload(payload);
 // Send Message
 secureTunnel->SendMessage(message);
 ```
+# Secure Tunnel Best Practices
+* You MUST NOT perform blocking operations on any callback, or you will cause a deadlock.
+* If you do not provide a Client Token during creation of the Secure Tunnel, one will be automatically generated for you to use in reconnections. This token is not saved outside of the current Secure Tunnel Client. If the Client is destroyed, the original access tokens must be rotated to connect to the secure tunnel again. Information on rotating tokens can be found here: https://docs.aws.amazon.com/iot/latest/developerguide/iot-secure-tunneling-troubleshooting.html
+* Client tokens MUST be unique. You cannot for example, pair a Client Token with an Access Token on one secure tunnel, and then use the same Client Token with a different Access Token on a separate secure tunnel. The Secure Tunnel Service will not allow reuse of a Client Token.
```

Comment:
Nitpick: `The Secure Tunnel Service will not allow reuse of a Client Token.` -> `The Secure Tunnel Service will not allow reuse of a Client Token for new secure tunnels. Client Tokens can only be reused when reconnecting` or something similar.